### ISSUE TITLE
Search for rides - API support

### ISSUE THREAD

**Issue by EvelynL09** [rise-summer/team2-studentRideSharing#6]
- Add Backend API support for GET /api/rides/cfrom=city&sfrom=state&cto=city&sto=state&time=x

**Comment by EvelynL09** [comment]
Should we utilize this feature of mongodb and also save addresses as coordinates for querying nearby locations?https://docs.mongodb.com/manual/geospatial-queries/ @micheyale @alex-dinh 
Also, what format of time you guys think is easier to passed as query param in the url "/api/rides/from=x&to=x&time=x"?
From the database side, anything that can generate a  [Date() ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) object in JS would work.

**Comment by micheyale** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Implement Search Ride Request with frontend](https://trello.com/c/8LfY6M4h/17-implement-search-ride-request-with-frontend)

**Comment by EvelynL09** [comment]
> Should we utilize this feature of mongodb and also save addresses as coordinates for querying nearby locations?https://docs.mongodb.com/manual/geospatial-queries/ @micheyale @alex-dinh 

Update: Geospacial query only allows query upon one pair of coordinates, and I'm currently only doing it for starting position.
Might consider doing something mentioned under this question [mongoDB GeoJSON, Query near two point, or line
](https://stackoverflow.com/a/60030799/12235978), which is manually calculate the distance based on the coordinates. Not sure if it's worth of doing it, and it might hurt search's performance. But if JS or React cannot perform such filtering at lower cost, I'll give it a try.

> Also, what format of time you guys think is easier to passed as query param in the url "/api/rides/from=x&to=x&time=x"?
> From the database side, anything that can generate a [Date() ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) object in JS would work.

Update: An elegant way that I found of doing it is by using the querystring package, as I did in commit https://github.com/rise-summer/team2-studentRideSharing/pull/7/commits/0dd4632163437553958681517ed24a4c5ba73e04.
